lib/src/fetch/config_fetcher.dart

```diff
@@ -185,17 +185,17 @@ class ConfigFetcher implements Fetcher {
         return FetchResponse.notModified();
       } else if (response.statusCode == 404 || response.statusCode == 403) {
         final error =
-            'Double-check your API KEY at https://app.configcat.com/apikey. Received unexpected response: ${response.statusCode} ${response.statusMessage}';
-        _errorReporter.error(error);
+            'Your SDK Key seems to be wrong. You can find the valid SDK Key at https://app.configcat.com/sdkkey. Received unexpected response: ${response.statusCode} ${response.statusMessage}';
+        _errorReporter.error(1100, error);
         return FetchResponse.failure(error, false);
       } else {
         final error =
-            'Unexpected HTTP response was received: ${response.statusCode} ${response.statusMessage}';
-        _errorReporter.error(error);
+            'Unexpected HTTP response was received while trying to fetch config JSON: ${response.statusCode} ${response.statusMessage}';
+        _errorReporter.error(1101, error);
         return FetchResponse.failure(error, true);
       }
     } catch (e, s) {
-      _errorReporter.error('Exception occurred during fetching.', e, s);
+      _errorReporter.error(1103, 'Unexpected error occurred while trying to fetch config JSON.', e, s);
```

Comment:
Request timeout (1102) is not handled here.  I'm not sure it's an oversight or is just not possible in Dart.